modules/aws-terraform-backend/iam.tf

```
@@ -1,4 +1,4 @@
-resource "aws_iam_role" "this" {
+resource "aws_iam_role" "admin_tfstate_role" {
```

Why? You can use `this` and `that` for 2 resources of the same type. Or in this case for example, `main` vs `aux`?
```suggestion
resource "aws_iam_role" "main" {
```